File: Top 100 Codes/05-Operations on Strings/22-Print all permutations of a given string in lexicographically sorted order/Printallpermutationsofagivenstringinlexicographicallsorted order.py
from itertools import permutations

def print_permutations(s):
    # Generate all permutations and convert to set to remove duplicates (if any)
    perm = sorted(set([''.join(p) for p in permutations(s)]))
    
    # Print each permutation
    for p in perm:
        print(p)

# Input from user
s = input("Enter a string: ")
print("All lexicographically sorted permutations:")
print_permutations(s)

File: Top 100 Codes/05-Operations on Strings/22-Print all permutations of a given string in lexicographically sorted order/Printallpermutationsofagivenstringinlexicographicallsortedorder.cpp
#include <iostream>
#include <algorithm>
using namespace std;

void printPermutations(string str) {
    // Sort the string in lexicographical order
    sort(str.begin(), str.end());

    // Print all permutations using do-while and next_permutation
    do {
        cout << str << endl;
    } while (next_permutation(str.begin(), str.end()));
}

int main() {
    string input;
    cout << "Enter a string: ";
    cin >> input;

    cout << "All lexicographically sorted permutations:\n";
    printPermutations(input);

    return 0;
}
